### PR TITLE
Update compat matrix to mention SASL

### PR DESCRIPTION
**Pull request by dedemorton**
@acchen97 @suyograo Changes for https://github.com/elastic/logstash/issues/6489.

Can you help me get the exact details nailed down?

### CODE REVIEW

**Comment by suyograo** [comment]
@dedemorton 5.x of LS supports Kerberos SASL as well. otherwise LGTM

File: lib/logstash/inputs/kafka.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@
 # |0.9       |2.0.0 - 2.3.x   | 3.x.x |SSL |Works with the old Ruby Event API (`event['product']['price'] = 10`)  
 # |0.9       |2.4.x - 5.x.x   | 4.x.x |SSL |Works with the new getter/setter APIs (`event.set('[product][price]', 10)`)
 # |0.10.0.x  |2.4.x - 5.x.x   | 5.x.x |SSL |Not compatible with the <= 0.9 broker
-# |0.10.1.x  |2.4.x - 5.x.x   | 6.x.x |SSL |Not compatible with <= 0.10.0.x broker
+# |0.10.1.x  |2.4.x - 5.x.x   | 6.x.x |SSL, Kerberos SASL |Not compatible with <= 0.10.0.x broker
```
Comment (acchen97) [inline lib/logstash/inputs/kafka.rb]: @suyograo I think we need another line here. How about something like this for the last three rows?

Kafka 0.10.0.x | Plugin v5.x.x | SSL, Kerberos SASL (v5.1.0+)
Kafka 0.10.1.0 | Plugin v6.0.0 - 6.2.1 | SSL, Kerberos SASL (v6.1.0+)
Kafka 0.10.1.1 | Plugin v6.2.2 - 6.x.x | SSL, Kerberos SASL

Comment (acchen97) [inline lib/logstash/inputs/kafka.rb]: The Kerberos SASL backported to 5.1.0 isn't in the change log. Was the backport done similarly in the output as well? If so, what plugin version?

**Comment by dedemorton** [comment]
@suyograo @acchen97 I've made some additional changes to reflect your comments. Hopefully this gets us closer.

File: lib/logstash/inputs/kafka.rb [hunk 2]
```diff
@@ -4,26 +4,34 @@
...
-# |0.9       |2.4.x - 5.x.x   | 4.x.x |SSL |Works with the new getter/setter APIs (`event.set('[product][price]', 10)`)
-# |0.10.0.x  |2.4.x - 5.x.x   | 5.x.x |SSL |Not compatible with the <= 0.9 broker
-# |0.10.1.x  |2.4.x - 5.x.x   | 6.x.x |SSL |Not compatible with <= 0.10.0.x broker
+# |Kafka Client Version |Logstash Version |Plugin Version |Why?
+# |0.8       |2.0.0 - 2.x.x   |<3.0.0 |Legacy, 0.8 is still popular 
+# |0.9       |2.0.0 - 2.3.x   | 3.x.x |Works with the old Ruby Event API (`event['product']['price'] = 10`)  
+# |0.9       |2.4.x - 5.x.x   | 4.x.x |Works with the new getter/setter APIs (`event.set('[product][price]', 10)`)
+# |0.10.0.x  |2.4.x - 5.x.x   | 5.x.x |Not compatible with the <= 0.9 broker
+# |0.10.1.0  |2.4.x - 5.x.x   | 6.0.0 - 6.2.1 |Not compatible with <= 0.10.0.x broker
+# |0.10.1.1  |2.4.x - 5.x.x   | 6.2.2 - 6.x.x |Not compatible with <= 0.10.1.0 broker
```
Comment (dedemorton) [inline lib/logstash/inputs/kafka.rb]: I'm not sure about this compatibility statement. Added the row and followed the convention in the table.

Comment (suyograo) [inline lib/logstash/inputs/kafka.rb]: we should remove the last line `|0.10.1.1  |2.4.x - 5.x.x   | 6.2.2 - 6.x.x |Not compatible with <= 0.10.1.0 broker`. This is not true.

Comment (suyograo) [inline lib/logstash/inputs/kafka.rb]: `0.10.1.1` is a point release which is compatible with  `0.10.1.0`

File: lib/logstash/inputs/kafka.rb [hunk 3]
```diff
@@ -4,26 +4,34 @@
...
 # |==========================================================
 # 
 # NOTE: We recommended that you use matching Kafka client and broker versions. During upgrades, you should
 # upgrade brokers before clients because brokers target backwards compatibility. For example, the 0.9 broker
 # is compatible with both the 0.8 consumer and 0.9 consumer APIs, but not the other way around.
 #
+# This input supports connecting to Kafka over:
```
Comment (suyograo) [inline lib/logstash/inputs/kafka.rb]: This is much better documented here.

**Comment by suyograo** [comment]
@dedemorton minor comment re: your question. Feel free to merge after that.

File: lib/logstash/inputs/kafka.rb (hunk 3 above)
Comment (acchen97) [inline lib/logstash/inputs/kafka.rb]: ++ I like it

File: lib/logstash/inputs/kafka.rb (hunk 2 above)
Comment (acchen97) [inline lib/logstash/inputs/kafka.rb]: @suyog I see. I had deduced that last line from this PR https://github.com/logstash-plugins/logstash-input-kafka/pull/160. 0.10.1.1 compatibility being enabled in 6.2.2.

Comment (acchen97) [inline lib/logstash/inputs/kafka.rb]: If we remove the last line, we should change line 20 to be 0.10.1.x instead of 0.10.1.0 as well.

Comment (dedemorton) [inline lib/logstash/inputs/kafka.rb]: @suyograo Ok, so just to confirm, line 20 should say:

`````
|0.10.1.x  |2.4.x - 5.x.x   | 6.x.x  | 
`````

and not mention compatibility.

Comment (dedemorton) [inline lib/logstash/inputs/kafka.rb]: And line 21 will be deleted. :-)

**Pull request closed without merging**